src/include/rho/AllocatorSuperblock.hpp

```diff
@@ -0,0 +1,156 @@
+/*
+ *  R : A Computer Language for Statistical Data Analysis
+ *  Copyright (C) 1995, 1996  Robert Gentleman and Ross Ihaka
+ *  Copyright (C) 1998-2007   The R Development Core Team.
+ *  Copyright (C) 2008-2014  Andrew R. Runnalls.
+ *  Copyright (C) 2014 and onwards the Rho Project Authors.
+ *
+ *  Rho is not part of the R project, and bugs and other issues should
+ *  not be reported via r-bugs or other R project channels; instead refer
+ *  to the Rho website.
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, a copy is available at
+ *  https://www.R-project.org/Licenses/
+ */
+
+#ifndef RHO_ALLOCATORSUPERBLOCK_HPP
+#define RHO_ALLOCATORSUPERBLOCK_HPP
+
+#include <cstdint>
+#include <functional>
+
+#include "GCNodeAllocator.hpp"
+
+typedef std::uint64_t u64;
+typedef std::uint32_t u32;
+
+using std::function;
+using std::size_t;
+
+/*
+ * A small object superblock has size 2^18.
+ * Minimum object size = 32 bytes.
+ */
+#define SUPERBLOCK_BITS (18)
```

Comment:
Instead of using #define, you can do something like:
  static constexpr unsigned SUPERBLOCK_BITS = 18;
etc. inside the class scope.